autonomic-administration-plugin/src/test/java/br/com/autonomiccs/autonomic/administration/plugin/AdministrationAgentTest.java

```diff
@@ -0,0 +1,665 @@
+        Assert.assertEquals(0, result.size());
+    }
+
+    @Test
+    public void createAllClustersResourcesUpTestEmptyClustersList() {
+        Mockito.doReturn(new ArrayList<>()).when(clusterService).listAllClusters();
+        Mockito.doReturn(new ArrayList<>()).when(administrationAgent).getClusterUpHosts(Mockito.any(ClusterVO.class));
+        List<ClusterResourcesUp> result = administrationAgent.createAllClustersResourcesUp();
+
+        verifyCreateAllClustersResourcesUpCalledMethods(0);
+        Assert.assertEquals(0, result.size());
+    }
+
+    @Test
+    public void mapAndExecuteVMsMigrationsTestEmptyHostList() {
+        ClusterVO cluster = createCluster(null);
+        List<HostResources> hosts = new ArrayList<>();
+        setupMapAndExecuteVMsMigrationsTest(hosts, true);
+
+        administrationAgent.mapAndExecuteVMsMigrations(cluster, clusterAdministrationHeuristicAlgorithm);
+
+        verifyMapAndExecuteVMsMigrationsCalledMethods(0, 0);
+    }
+
+    @Test
+    public void mapAndExecuteVMsMigrationsTestOneHost() {
+        ClusterVO cluster = createCluster(null);
+        List<HostResources> hosts = createHosts(1);
+        setupMapAndExecuteVMsMigrationsTest(hosts, true);
+
+        administrationAgent.mapAndExecuteVMsMigrations(cluster, clusterAdministrationHeuristicAlgorithm);
+
+        verifyMapAndExecuteVMsMigrationsCalledMethods(0, 0);
+    }
+
+    @Test
+    public void mapAndExecuteVMsMigrationsTestFullExecution() {
+        ClusterVO cluster = createCluster(null);
+        List<HostResources> hosts = createHosts(2);
+        setupMapAndExecuteVMsMigrationsTest(hosts, true);
+
+        administrationAgent.mapAndExecuteVMsMigrations(cluster, clusterAdministrationHeuristicAlgorithm);
+
+        verifyMapAndExecuteVMsMigrationsCalledMethods(1, 1);
+    }
+
+    @Test
+    public void mapAndExecuteVMsMigrationsTestEmptyMigrationsMap() {
+        ClusterVO cluster = createCluster(null);
+        List<HostResources> hosts = createHosts(2);
+        setupMapAndExecuteVMsMigrationsTest(hosts, false);
+
+        administrationAgent.mapAndExecuteVMsMigrations(cluster, clusterAdministrationHeuristicAlgorithm);
+
+        verifyMapAndExecuteVMsMigrationsCalledMethods(1, 0);
+    }
+
+    @Test
+    public void migrateVMTestTargetHostEqualsVmHost() throws Exception {
+        VMInstanceVO vmInstanceVo = migrateVMSetupVmInstanceVo(1l);
+        HostVO hostVo = migrateVMSetupHostVo(1l);
+        Mockito.doReturn(vmInstanceVo).when(userVmService).migrateVirtualMachine(Mockito.anyLong(), Mockito.any(HostVO.class));
+        Mockito.doReturn(new HostVO("")).when(hostService).findHostById(Mockito.anyLong());
+
+        administrationAgent.migrateVM(vmInstanceVo, hostVo);
+
+        Mockito.verify(userVmService, Mockito.times(0)).migrateVirtualMachine(Mockito.anyLong(), Mockito.any(HostVO.class));
+    }
+
+    @Test
+    public void migrateVMTestTargetHostDifferentFromVmHost() throws Exception {
```

Comment:
For this method there is the flow of execution when an exception happens that should be tested.